Source/WebCore/dom/TrustedType.cpp

```diff
@@ -207,4 +210,28 @@ ExceptionOr<String> requireTrustedTypesForPreNavigationCheckPasses(ScriptExecuti
         : nullString());
 }
 
+ExceptionOr<RefPtr<Text>> processNodeOrStringAsTrustedType(Ref<Document> document, RefPtr<Node> parent, std::variant<RefPtr<Node>, String, RefPtr<TrustedScript>> variant)
+{
+    RefPtr<Text> text;
+    if (std::holds_alternative<String>(variant))
+        text = Text::create(document, WTFMove(std::get<String>(variant)));
```

Comment:
@rwlbuis it's not quite as the spec is written (because WebKit isn't quite how the spec is written already) but this line should be effectively doing step 2.

And then the text node is processed below. Which handles Original Strings and Original Text Nodes in one go.